src/web/client/utility/UrlBuilder.ts

```diff
@@ -3,34 +3,61 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  */
 
-import { httpMethod, MULTI_ENTITY_URL_KEY, pathParamToSchema, SINGLE_ENTITY_URL_KEY } from "../common/constants";
+import { entityAttributesWithBase64Encoding, httpMethod, schemaEntityKey, schemaEntityName, schemaKey } from "../common/constants";
 import PowerPlatformExtensionContextManager from "../common/localStore";
+import { getEntity } from "./schemaHelper";
 
 export const getParameterizedRequestUrlTemplate = (isSingleEntity: boolean) => {
     const powerPlatformContext = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext();
     if (isSingleEntity) {
-        return powerPlatformContext.dataSourcePropertiesMap.get(SINGLE_ENTITY_URL_KEY) as string;
+        return powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.SINGLE_ENTITY_URL) as string;
     }
 
-    return powerPlatformContext.dataSourcePropertiesMap.get(MULTI_ENTITY_URL_KEY) as string;
+    return powerPlatformContext.dataSourcePropertiesMap.get(schemaKey.MULTI_ENTITY_URL) as string;
 };
 
-export function getRequestURL(dataverseOrgUrl: string, entity: string, entityId: string, method: string, isSingleEntity: boolean): string {
+export function getRequestURL(
+    dataverseOrgUrl: string,
+    entity: string,
+    entityId: string,
+    method: string,
+    isSingleEntity: boolean,
+    attributeQueryParameters?: string): string {
     const powerPlatformContext = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext();
     let parameterizedUrlTemplate = getParameterizedRequestUrlTemplate(isSingleEntity);
+
     switch (method) {
         case httpMethod.GET:
             parameterizedUrlTemplate = parameterizedUrlTemplate
-                + powerPlatformContext.entitiesSchemaMap.get(pathParamToSchema.get(entity) as string)?.get('_fetchQueryParameters');
+                + (attributeQueryParameters ?? getEntity(entity)?.get(schemaEntityKey.FETCH_QUERY_PARAMETERS));
             break;
         default:
             break;
     }
 
-    return parameterizedUrlTemplate.replace('{dataverseOrgUrl}', dataverseOrgUrl).replace('{entity}', entity)
-        .replace('{entityId}', entityId).replace('{api}', powerPlatformContext.dataSourcePropertiesMap.get('api') as string)
-        .replace('{data}', powerPlatformContext.dataSourcePropertiesMap.get('data') as string)
-        .replace('{version}', powerPlatformContext.dataSourcePropertiesMap.get('version') as string);
+    return parameterizedUrlTemplate.replace('{dataverseOrgUrl}', dataverseOrgUrl).replace('{entity}', getEntity(entity)?.get(schemaEntityKey.DATAVERSE_ENTITY_NAME) as string)
```

Comment:
Can we write a formatString utility instead of chaining replace across code?

Reply:
Agree, we need to cleanup this code. Out of scope for this PR.